docs/proposals/locking_notes.md

```diff
@@ -0,0 +1,62 @@
+# Capture Locking Notes during Locks creation and retrieve same during List Locks.
+
+## Concept
+
+The git-lfs REST API for Locks creation should be able to accept additonal attribute (message/notes) which would be easy to have some information related to lock creation. This same information can be retrieved back with the git-lfs List Locks REST API.
+
+  - Allow to have additional attribute to store the lock message/notes during lock creation
+  - Include lock message/notes in the git-lfs List Locks REST API response
+
+## API extensions
+
+The current Git LFS File Locking API [https://github.com/git-lfs/git-lfs/blob/v3.2.0/docs/api/locking.md] doesn't have a field to capture some information related to Locks creation which would be useful to understand why and from where the lock was acquired. To get more understanding, below is the use case:
+
+In our application team development feature, there are 3 workspaces with 2 users working on a single Repository containing 6 files.
+
+Here the User1 is working on FileA, FileB under Workspace W1, User2 is working on FileC, FileD under Workspace W2 and also User1 is working on FileE, FileF under Workspace W3.
```

Comment:
I think I'd like to phrase this as a generic comment field rather than being used for any particular workflow.  It may be that you want to use it for this purpose, but others may simply be interested in a free-form comment field, much like a commit message.

Reply:
To be clearer, I think we should avoid the discussion of your particular use case.  It is one possible use case, but it isn't a particularly common one, and I think that now we have the Concept section and the “With this enhancement” section below, it should be clear to folks why a general-purpose comment field is useful and would be valuable.